recipes/microprofile/all/conanfile.py

```diff
@@ -0,0 +1,222 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class MicroprofileConan(ConanFile):
+    name = "microprofile"
+    license = "Unlicense"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jonasmr/microprofile"
+    description = "Microprofile is a embeddable profiler in a few files, written in C++"
+    topics = ("conan", "profiler", "embedded", "timer")
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enabled": [True, False],
+        "debug": [True, False],
+        "use_miniz": [True, False],
+        "thread_buffer_size": "ANY",
+        "thread_gpu_buffer_size": "ANY",
+        "max_frame_history": "ANY",
+        "webserver_port": "ANY",
+        "webserver_maxframes": "ANY",
+        "webserver_socket_buffer_size": "ANY",
+        "gpu_frame_delay": "ANY",
+        "name_max_length": "ANY",
+        "max_timers": "ANY",
+        "max_threads": "ANY",
+        "max_string_length": "ANY",
+        "timeline_max_tokens": "ANY",
+        "thread_log_frames_reuse": "ANY",
+        "max_groups": "ANY",
+        "use_big_endian": [True, False],
+        "enable_gpu_timer_callbacks": [True, False],
+        "enable_gl_timers": [True, False],
+        "enable_d3d11_timers": [True, False],
+        "enable_d3d12_timers": [True, False],
+        "enable_vulkan_timers": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "enabled": True,
+        "debug": False,
+        "use_miniz": False,
+        "thread_buffer_size": 2048 << 10,
+        "thread_gpu_buffer_size": 128 << 10,
+        "max_frame_history": 512,
+        "webserver_port": 1338,
+        "webserver_maxframes": 30,
+        "webserver_socket_buffer_size": 16 << 10,
+        "gpu_frame_delay": 5,
+        "name_max_length": 64,
+        "max_timers": 1024,
+        "max_threads": 32,
+        "max_string_length": 128,
+        "timeline_max_tokens": 64,
+        "thread_log_frames_reuse": 200,
+        "max_groups": 128,
+        "use_big_endian": False,
+        "enable_gpu_timer_callbacks": False,
+        "enable_gl_timers": False,
+        "enable_d3d11_timers": False,
+        "enable_d3d12_timers": False,
+        "enable_vulkan_timers": False
```

Comment:
```suggestion
        "enable_timer": [None, "gl", "d3d11", "d3d12", "vulkan"]
```

It can be simplified as they mutual exclusive. Some options are only allowed on windows, so you still can validated.